msticpy/context/azure/sentinel_core.py

```diff
@@ -54,97 +111,102 @@ class MicrosoftSentinel(
-        self.default_subscription: Optional[str] = None
-        self._resource_id = res_id
-        self._default_resource_group: Optional[str] = None
         self.sent_urls: Dict[str, str] = {}
         self.sent_data_query: Optional[SentinelQueryProvider] = None  # type: ignore
         self.url: Optional[str] = None
         self._token: Optional[str] = None
 
-        workspace = kwargs.get("workspace", ws_name)
-        self._default_workspace: Optional[str] = workspace
-        self.workspace_config = WorkspaceConfig(workspace)
-
         logger.info("Initializing Microsoft Sentinel connector")
         logger.info(
-            "Params: Cloud=%s; ResourceId=%s; Workspace=%s",
+            "Params: Cloud=%s; ResourceId=%s; Workspace=%s, Subscription=%s, ResourceGroup=%s",
             self.cloud,
-            self._resource_id,
-            workspace,
+            self.default_resource_id,
+            self.default_workspace_name,
+            self.default_subscription_id,
+            self.default_resource_group,
         )
 
-        if self._resource_id:
-            # If a resource ID is supplied, use that
-            logger.info("Initializing from resource ID")
-            self.url = self._build_sent_paths(
-                self._resource_id, self.base_url
-            )  # type: ignore
-            res_id_parts = parse_resource_id(self._resource_id)
-            self.default_subscription = res_id_parts["subscription_id"]
-            self._default_resource_group = res_id_parts["resource_group"]
-            self._default_workspace = workspace or res_id_parts["workspace_name"]
-            if self._default_workspace in WorkspaceConfig.list_workspaces():
-                self.workspace_config = WorkspaceConfig(
-                    workspace=self._default_workspace
-                )
-                logger.info("Workspace settings found for %s", self._default_workspace)
-        else:
-            # Otherwise - use details from specified workspace or default from settings
-            logger.info("Initializing from workspace settings")
-            self.default_subscription = self.workspace_config.get(
-                "subscription_id", sub_id
-            )
-            self._default_resource_group = self.workspace_config.get(
-                "resource_group", res_grp
-            )
-            workspace_name = self.workspace_config.get("workspace_name", workspace)
-            self._resource_id = self._build_sent_res_id(
-                sub_id=self.default_subscription,
-                res_grp=self._default_resource_group,
-                ws_name=workspace_name,
-            )
-            logger.info("Resource ID set to %s", self._resource_id)
-            self._default_workspace = workspace_name
-            self.url = self._build_sent_paths(
-                self._resource_id, self.base_url  # type: ignore
+        if not any(ws_param in init_kwargs for ws_param in _WS_PARAMETERS):
```

Comment:
I feel like this block should be in the `_get_ws_defaults` or `_set_ws_defaults` functions somewhere.